### PR TITLE
add lock for static variable operation

### PR DESCRIPTION
**Pull request by xufang-lisa**
### Details:
 - *add lock for static variable operation*

### Tickets:
 - *CVS-129063*

### CODE REVIEW

**Comment by liubo-intel** [comment]
Hi, @xufang-lisa : could you please provide more information about why we need add mutex lock here?

File: src/plugins/intel_cpu/src/nodes/conv.cpp [hunk 1]
```diff
@@ -371,6 +371,8 @@ const std::vector<impl_desc_type>& Convolution::getDefaultImplPriority() {
         impl_desc_type::ref,
     };
 
+    static std::mutex mutex;
+    std::unique_lock<std::mutex> lock(mutex);
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:375]: Modify static variable is not thread-safe, it makes sense to add lock.

**Comment by liubo-intel** [comment]
Hi, @maxnick : I found you have some pre-analysis about this ticket. do you have any comments about this change? I mean is there any risk for model compile performance if add  mutex lock here? 
if `const` is must needed here for this `priorities` variable, do you think it would be better that we initialize it by `isBrgConvAvailable()` condition(I mean if isBrgConvAvailable()==true, we initialize this `priorities`  list with `impl_desc_type::brgconv`, others w/o) before brgconv is fully used for all platforms?

**Comment by maxnick** [comment]
> Hi, @maxnick : I found you have some pre-analysis about this ticket. do you have any comments about this change? I mean is there any risk for model compile performance if add mutex lock here? if `const` is must needed here for this `priorities` variable, do you think it would be better that we initialize it by `isBrgConvAvailable()` condition(I mean if isBrgConvAvailable()==true, we initialize this `priorities` list with `impl_desc_type::brgconv`, others w/o) before brgconv is fully used for all platforms?

Thank you for rising this concern!!! Using a look is indeed an overkill and will definitely cause unnecessary system calls  and thread back offs during model compilation, which will impact the loading time. But, this problem may be resolved relatively easy:
```cpp
const std::vector<impl_desc_type>& Convolution::getDefaultImplPriority() {
    const static auto priorities = [] {
        std::vector<impl_desc_type> priorities = {
            impl_desc_type::unknown,
            impl_desc_type::dw_acl,
            impl_desc_type::winograd_acl,
            impl_desc_type::gemm_acl,
            impl_desc_type::acl,
            impl_desc_type::brgconv_avx512_amx_1x1,
            impl_desc_type::brgconv_avx512_amx,
            impl_desc_type::jit_avx512_amx_dw,
            impl_desc_type::jit_avx512_amx_1x1,
            impl_desc_type::jit_avx512_amx,
            impl_desc_type::brgconv_avx512_1x1,
            impl_desc_type::brgconv_avx512,
            impl_desc_type::jit_uni_dw,
            impl_desc_type::jit_uni_1x1,
            impl_desc_type::jit_uni,
            impl_desc_type::jit_avx512_dw,
            impl_desc_type::jit_avx512_1x1,
            impl_desc_type::jit_avx512,
            impl_desc_type::brgconv_avx2_1x1,
            impl_desc_type::brgconv_avx2,
            impl_desc_type::jit_avx2_dw,
            impl_desc_type::jit_avx2_1x1,
            impl_desc_type::jit_avx2,
            impl_desc_type::jit_avx_dw,
            impl_desc_type::jit_avx_1x1,
            impl_desc_type::jit_avx,
            impl_desc_type::jit_sse42_dw,
            impl_desc_type::jit_sse42_1x1,
            impl_desc_type::jit_sse42,
            impl_desc_type::gemm_any,
            impl_desc_type::gemm_blas,
            impl_desc_type::gemm_avx512,
            impl_desc_type::gemm_avx2,
            impl_desc_type::gemm_avx,
            impl_desc_type::gemm_sse42,
            impl_desc_type::jit_gemm,
            impl_desc_type::ref_any,
            impl_desc_type::ref,
        };
        if (!isBrgConvAvailable()) {
            priorities.erase(std::remove_if(priorities.begin(),
                        priorities.end(),
                        [](impl_desc_type type) {
                            return type & impl_desc_type::brgconv;
                        }),
            priorities.end());
        }
   
        return priorities;
    }();

    return priorities;
}
```
So the whole initialization lambda is called once, thanks to the static duration of the `isBrgConvAvailable()` result.

File: src/plugins/intel_cpu/src/nodes/conv.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/plugins/intel_cpu/src/nodes/conv.cpp:375]: maybe we can use `std::call_once`?

**Review by riverlijunjie — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by sunxiaoxia2022 — APPROVED** [review]

**Review by maxnick — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**